im-sdk-core/src/main/java/com/easemob/im/server/model/EMUser.java

```diff
@@ -17,13 +17,22 @@
 
     private final Boolean canLogin;
 
+    // keep this for backwards compatibility
+    @Deprecated
     public EMUser(String username, Boolean canLogin) {
         super(EntityType.USER);
         super.id(username);
         this.username = username;
         this.canLogin = canLogin;
     }
 
+    public EMUser(String username, String uuid, Boolean canLogin) {
+        super(EntityType.USER);
+        super.id(uuid);
```

Comment:
Add a uuid field to EMUser, not EMEntity.
It should be `super.id(username)` since the id of EMUser is the username.

Reply:
done